python/tests/test_util.py

```diff
@@ -99,6 +99,19 @@ def test_overflow(self):
                     pickle.dumps(target),
                     pickle.dumps(util.safe_np_int_cast(np.array([good_node]), dtype)))
 
+    def test_nonrectangular_intput(self):
+        bad_inputs = [
+                [0, 1, [2]],
+                [[0, 1, 2], []],
+                [(0, 1, 2), [2, 3]],
+                [(0, 1, 2), tuple()],
+                [(0, 1, 2), (2, )],
+                [(0, 1, 2), [2, 3]]]
```

Comment:
can you think of any other wierd inputs we might want to check for here, @hyanwong ?

Reply:
How about a (Python) array of unequal length 1D numpy arrays?